apps/full-site-editing/full-site-editing-plugin/mailerlite/subscriber-popup.php

```diff
@@ -0,0 +1,128 @@
+<?php //phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+namespace A8C\FSE\Mailerlite;
+
+/**
+ * Mailerlite widget class
+ * Display a subscriber popup for Mailerlite.
+ */
+class WPCOM_Widget_Mailerlite extends \WP_Widget {
+
+	/**
+	 * WPCOM_Widget_Mailerlite constructor.
+	 */
+	public function __construct() {
+		parent::__construct(
+			'wpcom-mailerlite',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Mailerlite subscriber popup', 'jetpack' ) ),
+			array(
+				'classname'                   => 'widget_mailerlite',
+				'description'                 => __( 'Display Mailerlite subscriber popup', 'jetpack' ),
+				'customize_selective_refresh' => true,
+			)
+		);
+	}
+
+	/**
+	 * Output the widget.
+	 *
+	 * @param array $args Display arguments including 'before_title', 'after_title', 'before_widget', and 'after_widget'.
+	 * @param array $instance - The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance ) {
+		/** This action is documented in modules/widgets/gravatar-profile.php */
+		do_action( 'jetpack_stats_extra', 'widget_view', 'mailerlite' );
+
+		if ( empty( $instance['account'] ) || empty( $instance['uuid'] ) ) {
+			if ( current_user_can( 'edit_theme_options' ) ) {
+				echo esc_js( $args['before_widget'] );
```

Comment:
I haven't seen this nor the `wp_kses` use here before – do you feel they're necessary? If an actor can set `before_widget`, they have admin access to the site, so they can do anything anyways. And I'm not sure what `wp_kses` is helping here.

Reply:
I don't feel it's necessary, but the `WordPress.Security.EscapeOutput.OutputNotEscaped` lint rule does :)
But I agree, ignored that rule.